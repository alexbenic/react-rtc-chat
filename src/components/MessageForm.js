import React, { Component } from "react";

class MessageForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      text: ""
    };
    this.props = props;
  }

  onSubmit = e => {
    e.preventDefault();
    this.props.onSubmit(this.state.text);
    this.setState({ text: "" });
  };

  onChange = e => {
    this.setState({ text: e.target.value });
    if (this.props.onChange) {
      this.props.onChange();
    }
  };

  render() {
    return (
      <div className="pt4 pb4 pl2 ma2 bt bw1 b--black-40">
        <form onSubmit={this.onSubmit} className="flex">
          <input
            type="text"
            className="flex-auto f5 fw5 bg-none b--none outline-0"
            placeholder="Type a message here then hit ENTER"
            onChange={this.onChange}
            value={this.state.text}
          />
        </form>
      </div>
    );
  }
}

export default MessageForm;
