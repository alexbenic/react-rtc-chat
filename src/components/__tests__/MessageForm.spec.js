import React from "react";
import each from "jest-each";
import { render, cleanup, fireEvent } from "react-testing-library";
import MessageForm from "../MessageForm";

afterEach(cleanup);

// prettier-ignore
each([
  "hey",
  "hello",
  "zdravo",
  "",
])
.test("form should submit message: %s", message => {
  const handleSubmit = jest.fn();
  const { container, getByPlaceholderText } = render(
    <MessageForm onSubmit={handleSubmit} />
  );

  const input = getByPlaceholderText("Type a message here then hit ENTER");
  input.value = message;

  fireEvent.change(input)
  fireEvent(container.querySelector("form"), new Event("submit"));

  expect(handleSubmit).toHaveBeenCalledTimes(1);
  expect(handleSubmit).toHaveBeenCalledWith(message);
});

test("message form snapshot", () => {
  const handleSubmit = jest.fn();
  const { container } = render(<MessageForm onSubmit={handleSubmit} />);

  expect(container).toMatchSnapshot();
});
