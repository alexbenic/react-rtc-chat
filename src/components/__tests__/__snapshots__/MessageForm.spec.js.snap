// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`message form snapshot 1`] = `
<div>
  <div
    class="pt4 pb4 pl2 ma2 bt bw1 b--black-40"
  >
    <form
      class="flex"
    >
      <input
        type="text"
        class="flex-auto f5 fw5 bg-none b--none outline-0"
        placeholder="Type a message here then hit ENTER"
        value=""
      />
    </form>
  </div>
</div>
`;
